src/app/shared/dynamic-form/components/form-lazy-options-base.component.ts

```diff
@@ -110,6 +110,9 @@ export class FormLazyOptionsBaseComponent extends FormBaseComponent implements O
       this.service[this.method](filters).subscribe({
         next: (response: any) => {
           this.config.options = response;
+          if (this.config.hasOtherOption) {
+            this.config.options.push({ key: 'OTHER', value: 'enum.PersonContactDetailType.OTHER' });
```

Comment:
You're hardcoding the key and value. I think this should be more generic than this.
As an example, for facilities i believe it's OTHER_FACILITY for the key.
Maybe instead of the boolean flag, we could have fallbackOptionKey and fallbackOptionValue?
Let's discuss this with the team.